src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +291,220 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
                 }
 
                 return default;
-            });
-    }
+            };
 
-    /// <summary>
-    /// Static members for <see cref="Driver{TCallable,TIn,TOut}"/>.
-    /// </summary>
-    internal static class Driver
-    {
-        // TODO: Define the aliases as constants.
+            command.AddValidator(validator);
+            return new[] { validator };
+        }
 
         /// <summary>
-        /// The subscription option.
+        /// Creates the simulate command.
         /// </summary>
-        internal static readonly OptionInfo<string> SubscriptionOption = new OptionInfo<string>(
-            ImmutableList.Create("--subscription"), "The subscription ID.");
+        /// <param name="entryPointCommands">The entry point commands that will be the sub commands to the created command.</param>
+        /// <returns>Tuple with the created simulate command and the validators for that command.</returns>
+        private static (Command, IEnumerable<ValidateSymbol<CommandResult>>) CreateSimulateCommand(
+            IEnumerable<(Command, IEnumerable<ValidateSymbol<CommandResult>>)> entryPointCommands)
```

Comment:
Have you considered swapping the command nesting, so that `simulate` and `submit` are sub-commands of the entry point command? I'm not sure if that would be better but it would mean that `simulate` and `submit` wouldn't both need the list of entry points.

Reply:
I've spent the day trying this out, and the code complexity wasn't noticeably different. From a user standpoint, my preference is to have it as the entry point command last, as that will match more closely to and English reading of the command line: `MyApp simulate MyEntryPoint` as opposed to `MyApp MyEntryPoint`. Subject being the name of the app, the verb being 'simulate' or 'submit' and the direct object being the entry point being simulated or submitted. Hence I'm going to keep it as the subject-verb-object ordering for the command hierarchy.